charts/cluster-scanner/templates/_helpers.tpl

```diff
@@ -0,0 +1,127 @@
+{{/*
+Expand the name of the chart.
+*/}}
+{{- define "cluster-scanner.name" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
+*/}}
+{{- define "cluster-scanner.fullname" -}}
+{{- if .Values.fullnameOverride }}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- $name := default .Chart.Name .Values.nameOverride }}
+{{- if contains $name .Release.Name }}
+{{- .Release.Name | trunc 63 | trimSuffix "-" }}
+{{- else }}
+{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{/*
+Create chart name and version as used by the chart label.
+*/}}
+{{- define "cluster-scanner.chart" -}}
+{{- printf "%s-%s" .Chart.Name .Chart.Version | replace "+" "_" | trunc 63 | trimSuffix "-" }}
+{{- end }}
+
+{{/*
+Common labels
+*/}}
+{{- define "cluster-scanner.labels" -}}
+helm.sh/chart: {{ include "cluster-scanner.chart" . }}
+{{ include "cluster-scanner.selectorLabels" . }}
+{{- if .Chart.AppVersion }}
+app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- end }}
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end }}
+
+{{/*
+Selector labels
+*/}}
+{{- define "cluster-scanner.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "cluster-scanner.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end }}
+
+{{/*
+Create the name of the service account to use
+*/}}
+{{- define "cluster-scanner.serviceAccountName" -}}
+{{- if .Values.serviceAccount.create }}
+{{- default (include "cluster-scanner.fullname" .) .Values.serviceAccount.name }}
+{{- else }}
+{{- default "default" .Values.serviceAccount.name }}
+{{- end }}
+{{- end }}
+
+{{/*
+Generates configmap data for mode-specific values
+*/}}
+{{- define "cluster-scanner.modeConfig" -}}
+{{- if eq .Values.global.scannerMode "local" }}
+rsi_mode: "sitting"
+{{- else if eq .Values.global.scannerMode "multi" }}
+rsi_mode: "mcm"
+{{- else }}
+{{- fail "invalid scannerMode, accepted options [local, multi]" }}
```

Comment:
This can be enforced at the chart level creating the proper `values.schema.json` without adding checks in the template (simplify templates and making them more readable)